File: =EXAMTIME/Unstop/19Aug/A.cpp
#include <bits/stdc++.h>
using namespace std;


int main() {
    int n;
    cin >> n;


    if (n == 1) {
        string singleString;
        cin >> singleString;
        cout << 1 << endl;
        return 0;
    }


    vector<string> arr(n);
    for (auto &v : arr)
        cin >> v;


    map<int, multiset<int>> mxHash;
    vector<int> hash(n);
    int ans = 0;
   
    for (int i = 0; i < n; i++) {
        for (auto v : arr[i]) {
            hash[i] ^= (1ll << (v - 'a'));
        }
        mxHash[hash[i]].insert((int)arr[i].size());
    }


    for (int i = 0; i < n; i++) {
        int hs = hash[i];
        mxHash[hs].erase(mxHash[hs].find((int)arr[i].size()));
        if (!mxHash[hs].empty()) {
            ans = max(ans, (int)arr[i].size() + (*--mxHash[hs].end()));
        }
        for (int mid = 0; mid < 26; mid++) {
            if (mxHash.count(hs ^ (1ll << mid)) and !mxHash[hs^(1ll<<mid)].empty()) {
                ans = max(ans, (int)arr[i].size() + (*--mxHash[hs ^ (1ll << mid)].end()));
            }
        }
        mxHash[hs].insert(arr[i].size());
    }


    cout << ans << endl;


    return 0;
}

File: =EXAMTIME/Unstop/19Aug/A.java
import java.util.*;

public class A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();

        if (n == 1) {
            String singleString = scanner.next();
            System.out.println(1);
            return;
        }

        String[] arr = new String[n];
        for (int i = 0; i < n; i++) {
            arr[i] = scanner.next();
        }

        Map<Integer, TreeSet<Integer>> mxHash = new HashMap<>();
        int[] hash = new int[n];
        int ans = 0;

        for (int i = 0; i < n; i++) {
            for (char c : arr[i].toCharArray()) {
                hash[i] ^= (1 << (c - 'a'));
            }
            if (!mxHash.containsKey(hash[i])) {
                mxHash.put(hash[i], new TreeSet<>());
            }
            mxHash.get(hash[i]).add(arr[i].length());
        }

        for (int i = 0; i < n; i++) {
            int hs = hash[i];
            mxHash.get(hs).remove(arr[i].length());
            if (!mxHash.get(hs).isEmpty()) {
                ans = Math.max(ans, arr[i].length() + mxHash.get(hs).last());
            }
            for (int mid = 0; mid < 26; mid++) {
                if (mxHash.containsKey(hs ^ (1 << mid)) && !mxHash.get(hs ^ (1 << mid)).isEmpty()) {
                    ans = Math.max(ans, arr[i].length() + mxHash.get(hs ^ (1 << mid)).last());
                }
            }
            mxHash.get(hs).add(arr[i].length());
        }

        System.out.println(ans);
    }
}
